harness/tests/cli/test_user.py

```diff
@@ -9,9 +10,89 @@
 from tests.cli import util
 
 
+def test_user_create_with_password_flag() -> None:
+    with util.standard_cli_rsps() as rsps:
+        userobj = bindings.v1User(active=True, admin=True, username="det-user", id=10017)
+        rsps.post(
+            "http://localhost:8080/api/v1/users",
+            status=200,
+            match=[
+                matchers.json_params_matcher(
+                    params={
+                        "isHashed": True,
+                        "user": {
+                            "username": "test-user-1",
+                        },
+                        "password": api.salt_and_hash("5DCAB140-f49b-4260-a451-fad6a10017ca"),
+                    },
+                    strict_match=False,
+                ),
+            ],
+            json={"user": userobj.to_json()},  # doesn't really match, but doesn't matter yet
+        )
+        cli.main(
+            ["user", "create", "test-user-1", "--password", "5DCAB140-f49b-4260-a451-fad6a10017ca"]
+        )
+
+
+def test_user_create_remote_account() -> None:
+    with util.standard_cli_rsps() as rsps:
+        userobj = bindings.v1User(active=True, admin=True, username="det-user", id=10018)
+        rsps.post(
+            "http://localhost:8080/api/v1/users",
+            status=200,
+            match=[
+                matchers.json_params_matcher(
+                    params={
+                        "isHashed": True,
+                        "user": {
+                            "username": "test-user-2",
+                            "remote": True,
+                        },
+                    },
+                    strict_match=False,
+                ),
+            ],
+            json={"user": userobj.to_json()},
+        )
+        cli.main(["user", "create", "test-user-2", "--remote"])
+
+
+@mock.patch("getpass.getpass")
+def test_user_create_interactive_password(mock_getpass: mock.MagicMock) -> None:
+    mock_getpass.side_effect = lambda *_: "8CBAAB59-21c5-45cb-b058-6e2f3ceaf03e"
+    with util.standard_cli_rsps() as rsps:
+        userobj = bindings.v1User(active=True, admin=True, username="det-user", id=10019)
+        rsps.post(
+            "http://localhost:8080/api/v1/users",
+            status=200,
+            match=[
+                matchers.json_params_matcher(
+                    params={
+                        "isHashed": True,
+                        "user": {
+                            "username": "test-user-3",
+                        },
+                        "password": api.salt_and_hash("8CBAAB59-21c5-45cb-b058-6e2f3ceaf03e"),
+                    },
+                    strict_match=False,
+                ),
+            ],
+            json={"user": userobj.to_json()},  # doesn't really match, but doesn't matter yet
+        )
+        cli.main(["user", "create", "test-user-3"])
+
+
+@mock.patch("getpass.getpass")
+def test_user_create_no_password_fails(mock_getpass: mock.MagicMock) -> None:
```

Comment:
This test makes more sense to me (though maybe you like the name `test_user_create_fails_with_empty_password`?).